persistent/test/main.hs

```diff
@@ -223,22 +225,13 @@ main = hspec $ do
             it "recognizes one line" $ do
                 parseLine "-- | this is a comment" `shouldBe`
                     Just
-                        ( Line 0
-                            [ DocComment "this is a comment"
-                            ]
+                        ( Line 0 $ pure
+                            (DocComment "this is a comment")
                         )
 
-            it "map parseLine" $ do
-                mapM parseLine ["Foo", "-- | Hello"]
-                    `shouldBe`
-                        Just
-                            [ Line 0 [Token "Foo"]
-                            , Line 0 [DocComment "Hello"]
-                            ]
```

Comment:
The switch to using `OverloadedLists` caused this to give an ambiguous type error:

```         
    • Couldn't match expected type ‘GHC.Exts.Item (t0 Line)’
                  with actual type ‘Line’
      The type variable ‘t0’ is ambiguous
    • In the expression: Line 0 [Token "Foo"]
      In the first argument of ‘Just’, namely
        ‘[Line 0 [Token "Foo"], Line 0 [DocComment "Hello"]]’
      In the second argument of ‘shouldBe’, namely
        ‘Just [Line 0 [Token "Foo"], Line 0 [DocComment "Hello"]]’
    |       
236 |                             [ Line 0 [Token "Foo"]
    |                               ^^^^^^^^^^^^^^^^^^^^        
```

I've opted to actually just delete this test, this function is being tested already with the other tests here (the one above and below it), and also in the context of multiple lines, in tests like this one:

https://github.com/yesodweb/persistent/blob/813fe422e8144afe76ac83f7fb94f158c496bc22/persistent/test/main.hs#L538

Reply:
Cool works for me!